invenio_app_ils/records_relations/api.py

```diff
@@ -256,13 +256,35 @@ def _validate_relation_between_records(self, first, second, relation_name):
             )
         )
 
-        if not (same_document or same_series or valid_edition):
+        valid_edition_fields = relation_name == "edition" and \
+            ('edition' in first and 'edition' in second) \
```

Comment:
Very minor, maybe an improvement?
```suggestion
            and first.get('edition', False) and second.get('edition', False)
```